### PR TITLE
Cythonize the construct

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

File: construct/adapters.pyx [hunk 1]
```diff
@@ -140,27 +163,32 @@ class StringAdapter(Adapter):
...
             if isinstance(self.encoding, str):
                 obj = obj.encode(self.encoding)
             else:
                 obj = self.encoding.encode(obj)
         return obj
+
     def _decode(self, obj, context):
         if not isinstance(obj, bytes):
-            obj = six.b("").join(obj)
+            obj = b("").join(obj)
```
Comment (arekbulski) [inline construct/adapters.pyx]: b"..." Is equivalent to b(...), Language construct is better than calling six

**Comment by arekbulski** [comment]
We should use cython as much as possible but I have few problems. (1) Is cython available for android apps? (2) Too much code at once, unreadable.

**Comment by mosquito** [comment]
Ok. It's merge request from my package cython-construct. You feel free when you want to decline it.

**Comment by arekbulski** [comment]
I appreciate it, and in fact I wanted to implement it myself. Just that its hard to comprehend 3000+ lines of code. :disappointed:

**Comment by mosquito** [comment]
More code moved from .py files. Actually I renamed only Cython-reserved keywords.

**Comment by arekbulski** [comment]
I am closing it for the time being. Cythonizing the code will have to wait until I am done with other architectural changes to the API.

**Pull request closed without merging**